src/awkward/contents/numpyarray.py

```diff
@@ -405,14 +405,14 @@ def _mergeable(self, other, mergebool):
                 return False
 
             if not matching_dtype and np.datetime64 in (
-                self._data.dtype,
-                other._data.dtype,
+                self._data.dtype.type,
+                other._data.dtype.type,
             ):
                 return False
 
             if not matching_dtype and np.timedelta64 in (
-                self._data.dtype,
-                other._data.dtype,
+                self._data.dtype.type,
+                other._data.dtype.type,
```

Comment:
This had always been wrong: it was considering all datetime types and timedelta types as mergeable, but they must not be. They should only be mergeable if they're exactly the the same dtype, including units.

The reason it was wrong was because `np.datetime64` and `np.timedelta64` are both type objects, like `self._data.dtype.type`, but unlike `self._data.dtype`, which is a dtype object. Sheesh!

Reply:
This is the only place where datetime mergeability is considered.